prow/tide/tide.go

```diff
@@ -200,7 +201,7 @@ func byRepoAndNumber(prs []PullRequest) map[string]PullRequest {
 // Note: an empty diff can be returned if the reason that the PR does not match
 // the TideQuery is unknown. This can happen happen if this function's logic
 // does not match GitHub's and does not indicate that the PR matches the query.
-func requirementDiff(pr *PullRequest, q *config.TideQuery) (string, int) {
+func requirementDiff(pr *PullRequest, q *config.TideQuery, cr *contextRegister) (string, int) {
```

Comment:
so contextRegister is composed of ignoreContext interfaces. If you want to register new ignoreContext interface you can can Register. What s the argument for this being an interface rather than a struct ?